### PR TITLE
Change to handle new format of sdk.conf.json.

### PR DESCRIPTION
**Pull request by molleti-vamshi**
Generate new format of metadata.json
Backward compatibility maintained

### CODE REVIEW

File: lib/scripts.js [hunk 1]
```diff
@@ -148,28 +168,33 @@ module.exports = function (gulp, gutil) {
...
             }
-          }
         }
       }
     }
 
     return gulp.src(['**/*.js', '!lib/**/*.js'], {cwd: src})
       .pipe(isWatch ? plumber()() : gutil.noop())
-      .pipe(uglify()({compress: false, mangle: false}))
+      .pipe(uglify()({compress: true, mangle: true}))
```
Comment (adamayres) [inline lib/scripts.js]: Why are we compressing this again?

Comment (molleti-vamshi) [inline lib/scripts.js]: Uglify here is used to get content in right format in downstream files. I have enabled compress and mangle to keep it consistent within the file. 

Change in variable names (mangle) doesn't cause errors downstream files.

There is a slight performance hit, which I think is valid for better readability.

File: lib/env-util.js [hunk 2]
```diff
@@ -48,35 +49,41 @@ gutil.env.filterFiles = function (filePaths) {
...
-  gutil.env.watchResIgnore = sdkConf.watchResIgnore || [];
-  gutil.env.verifyPlugin = sdkConf.hasOwnProperty('verifyPlugin') ? sdkConf.verifyPlugin : true;
+    if (is.object(sdkConf.ng) && is.not.empty(sdkConf.ng)) {
+        gutil.env.ng = {
+            module: sdkConf.ng.module || 'li',
+            textProperties: getSourcePaths(sdkConf.ng.addTextProperties || [])
+                .concat(['src/directives', 'src/services']),
+            moduleDependencyMap: sdkConf.ng.moduleDependencies || {},
+            moduleDependencies: getSourcePaths(
+                (function () {
```
Comment (adamayres) [inline lib/env-util.js]: Why the self-invoking function here?

File: lib/env-util.js [hunk 3]
```diff
@@ -48,35 +49,41 @@ gutil.env.filterFiles = function (filePaths) {
...
-    };
-  } else {
-    gutil.env.ng = false;
-  }
-  gutil.env.gitStatusVersion = sdkConf.gitStatusVersion || false;
-  gutil.env.serverConfig = sdkConf.serverConfig;
-  gutil.env.bowerComponentsPath = BOWER_COMPONENTS_PATH;
-  gutil.env.watchResIgnore = sdkConf.watchResIgnore || [];
-  gutil.env.verifyPlugin = sdkConf.hasOwnProperty('verifyPlugin') ? sdkConf.verifyPlugin : true;
+    if (is.object(sdkConf.ng) && is.not.empty(sdkConf.ng)) {
```
Comment (adamayres) [inline lib/env-util.js]: Indent here should be three spaces, not two. I suspect this is why the whole block appears to have changed (in the diff view) even though it didn't. Please put back to two spaces.

File: lib/scripts.js [hunk 4]
```diff
@@ -148,27 +168,33 @@ module.exports = function (gulp, gutil) {
...
+                }
             }
-          }
         }
       }
     }
 
     return gulp.src(['**/*.js', '!lib/**/*.js'], {cwd: src})
       .pipe(isWatch ? plumber()() : gutil.noop())
+      // Use Uglify to get the files in required format to be used in downstream function(createDepsMetadata)
```
Comment (adamayres) [inline lib/scripts.js]: Can you be more specific here, what is the format that we need the file in for it to work downstream?

**Comment by molleti-vamshi** [comment]
Change code to handle latest format of plugin metadata.

**Pull request merged**